tools/toolchains/__init__.py

```diff
@@ -1111,6 +1092,42 @@ def get_config_header(self):
         self.config_processed = True
         return self.config_file
 
+    @staticmethod
+    def generic_check_executable(tool_key, executable_name, levels_up,
+                                 nested_dir=None):
+        """
+        Positional args:
+        tool_key: the key to index TOOLCHAIN_PATHS
```

Comment:
Guess it is an argument for clarity/readability. I feel like it looks better this way from each respective toolchain point of view, and this function shows where that path comes from. Just passing in ``**["ARM", "armcc", 2, "bin"]`` looks better to me than ``**[TOOLCHAIN_PATHS["ARM"], "armcc", 2, "bin"]``